emgapi/views.py

```diff
@@ -1332,9 +1337,37 @@ def list(self, request):
     def create(self, request):
         try:
             response = requests.post(settings.GENOME_SEARCH_PROXY, data=request.data)
-            return HttpResponse(response.text, content_type='application/json')
         except requests.exceptions.RequestException:
+            logging.error(f'Failed to talk to genome search backend at {settings.GENOME_SEARCH_PROXY}')
             raise Http404('Genome search failed. Please try later.')
+        try:
+            response = response.json()
+        except (JSONDecodeError, ValueError):
+            logging.error(f'Failed to decode JSON from genome search backend')
+            logging.error(response.text)
+            raise Http404('Genome search failed. Please try later.')
+
+        results = response.get('results', [])
+        logging.info(f'Got {len(results)} search results')
+        mgyg_matches = [self.genome_id_from_accession(result.get('genome')) for result in results]
+        genomes = emg_models.Genome.objects.filter(genome_id__in=mgyg_matches).all()
```

Comment:
We discussed over slack, this could be a performance bottleneck. But, we will work out this when we start working on the performance of API.